src/buffalonwb/conversion_module.py

```diff
@@ -81,82 +78,110 @@ def conversion_function(source_paths, f_nwb, metafile, skip_raw=True, skip_proce
+        # Write raw data to NWB file
+        print('Writing to file: ' + out_file_raw)
         with NWBHDF5IO(out_file_raw, mode='w') as io:
-            print('Writing to file: ' + out_file_raw)
             io.write(nwbfile)
-            print(nwbfile)
+        print(nwbfile)
 
     if skip_processed:
-        print("skipping processed data...")
+        print("Skipping processed data...")
     if not skip_processed:
-        if no_copy or skip_raw:
+        if not copy_raw or skip_raw:
             nwbfile_proc = nwbfile
         else:
-            # copy from raw to maintain file linkage
+            # Copy from raw data NWB file to maintain file linkage
+            print('Copying NWB file ' + out_file_raw)
             raw_io = NWBHDF5IO(out_file_raw, mode='r')
             raw_nwbfile_in = raw_io.read()
             nwbfile_proc = raw_nwbfile_in.copy()
-            electrode_table_region = nwbfile_proc.acquisition['raw_ephys'].electrodes
-            print('Copying NWB file ' + out_file_raw)
+            electrode_table_region = nwbfile_proc.acquisition.electrodes
 
-        # BEHAVIOR (PROCESSED)
+        # Add processed behavior data
         if behavior_file is not None:
             add_behavior(
                 nwbfile=nwbfile_proc,
                 behavior_file=str(behavior_file),
                 metadata_behavior=metadata['Behavior']
             )
 
-        # PROCESSED COMPONENTS
-        # UNITS
+        # Add sorted units
         if sorted_spikes_nex5_file is not None:
             add_units(
                 nwbfile=nwbfile_proc,
                 nex_file_name=sorted_spikes_nex5_file
             )
 
-        # LFP
+        # Add LFP
         if lfp_mat_path is not None:
             add_lfp(
                 nwbfile=nwbfile_proc,
                 lfp_path=lfp_mat_path,
                 electrodes=electrode_table_region,
-                iterator_flag=lfp_iterator_flag,
+                iterator_flag=not no_lfp_iterator,
                 all_electrode_labels=electrode_labels
             )
 
-        # WRITE PROCESSED
-        if no_copy or skip_raw:
+        # Write processed data to NWB file
+        print('Writing to file: ' + out_file_processed)
+        if not copy_raw or skip_raw:
             with NWBHDF5IO(out_file_processed, mode='w') as io:
-                print('Writing to file: ' + out_file_processed)
                 io.write(nwbfile_proc)
-                print(nwbfile)
         else:
             with NWBHDF5IO(out_file_processed, mode='w', manager=raw_io.manager) as io:
-                print('Writing to file: ' + out_file_processed)
                 io.write(nwbfile_proc)
-                print(nwbfile)
-        if 'raw_io' in locals():
             raw_io.close()
 
+        print(nwbfile_proc)
+
+
+def check_source_paths(source_paths):
+    raw_nlx_path = None
+    lfp_mat_path = None
+    behavior_file = None
+    sorted_spikes_nex5_file = None
+    for k, v in source_paths.items():
```

Comment:
Not a big deal but it's a bit odd to iterate over the dictionary here. It would be more natural to do e.g. `if 'raw Nlx' in source_paths:`

Reply:
I agree - that's odd (I didn't write it). That is changed and cleaner now.